smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -437,23 +453,105 @@ private void updateResponsesMapWithResponseStatusAndObject(
         Shape operationOrError = shape.hasTrait(ErrorTrait.class) ? shape : operation;
         String statusCode = context.getOpenApiProtocol().getOperationResponseStatusCode(context, operationOrError);
         ResponseObject response = createResponse(
-                context, bindingIndex, eventStreamIndex, statusCode, operationOrError);
+                context, bindingIndex, eventStreamIndex, statusCode, operationOrError, false);
+
         responses.put(statusCode, response);
     }
 
+    private void updateResponsesMapWithResponseStatusAndMultipleObjects(
+            Context<T> context,
+            HttpBindingIndex bindingIndex,
+            EventStreamIndex eventStreamIndex,
+            List<StructureShape> shapes,
+            Map<String, ResponseObject> responses
+    ) {
+        List<ResponseObject> errors = new ArrayList<>();
+        ResponseObject.Builder combinedResponse = ResponseObject.builder();
+        String statusCode = context.getOpenApiProtocol().getOperationResponseStatusCode(context, shapes.get(0));
+        OpenApiConfig.ErrorStatusConflictHandlingStrategy strategy = context.getConfig().getOnErrorStatusConflict();
+        Map<String, List<MediaTypeObject>> contents = new LinkedHashMap<>();
+
+        // make response object for each error response & add to list
+        for (StructureShape shape : shapes) {
+            errors.add(createResponse(context, bindingIndex, eventStreamIndex, statusCode, shape,
+                    true));
+        }
```

Comment:
You can use streams for this to make this code more compact and idiomatic, e.g.,

```
        List<ResponseObject> errors = shapes.stream()
                .map(shape -> createResponse(context, bindingIndex, eventStreamIndex, statusCode, shape, true))
                .collect(Collectors.toList());

```

Reply:
for loop kept, following discussion.